lib/token-assert.js

```diff
@@ -102,19 +103,16 @@ TokenAssert.prototype.sameLine = function(options) {
         return;
     }
 
-    if (token.loc.end.line !== nextToken.loc.start.line) {
-        nextToken.whitespaceBefore = '';
-        this.emit('error', {
-            message: options.message || token.value + ' and ' + nextToken.value + ' should be on the same line',
-            line: token.loc.end.line,
-            column: token.loc.end.column,
-            fixed: true
-        });
-    }
+    this.linesBetween({
+        token: token,
+        nextToken: nextToken,
+        message: options.message || token.value + ' and ' + nextToken.value + ' should be on the same line',
+        exactly: -1
```

Comment:
I think it's a special case we shouldn't check here. Probably `sameLine: true` would be more informative. Or just a separate method to check that. Or rename of `linesBetween` to `newLinesBetween` ;-(